airbyte-cdk/bulk/toolkits/extract-cdc/src/main/kotlin/io/airbyte/cdk/read/cdc/CdcPartitionsCreator.kt

```diff
@@ -41,54 +41,79 @@ class CdcPartitionsCreator<T : Comparable<T>>(
     }
 
     override suspend fun run(): List<PartitionReader> {
-        if (CDCNeedsRestart) {
+        if (reset.get()) {
+            feedBootstrap.stateQuerier.resetFeedStates()
```

Comment:
Moving this here from where it was should make no functional difference but it will make it easier for me to make changes to this logic in the very near future.